mlir/lib/Dialect/EmitC/IR/EmitC.cpp

```diff
@@ -227,9 +235,11 @@ bool CastOp::areCastCompatible(TypeRange inputs, TypeRange outputs) {
   Type input = inputs.front(), output = outputs.front();
 
   return ((llvm::isa<IntegerType, FloatType, IndexType, emitc::OpaqueType,
```

Comment:
Can we use `return emitc::isIntegerIndexOrOpaqueType(input) || emitc::isSupportedFloatType(input) || isa<emitc::PointerType>(input);` here and for `output`?